### DISCUSSION TITLE
Is there a best-practices/overview comparing each of the supported anomaly detection models?

### DISCUSSION THREAD

**Discussion by SilverStarCoder** [open-edge-platform/anomalib#1785] [Q&A]
There are many supported anomaly detection models in anomalib, but I don't really know which one to apply to a specific usecase. Is there a comparison between all of them showing pros and cons of each?  Something like,  "if images are large in size, best to use X model, if no masks are available, use Y model, etc..." would be very helpful.  Doing trial and error is becoming very time consuming for my custom datasets.

**Comment by abc-125** [comment]
From my experience in industrial anomaly detection, I would suggest trying EfficientAD as the first choice. It is the fastest model available with high performance. If you don't have many images for training (100 or less), you can try PatchCore, it works really well for smaller datasets and it trains faster than EfficientAD. Also, you can check this [paper](https://arxiv.org/pdf/2303.14535.pdf), appendix F, to see the qualitative comparison of several different models. I hope this helps.

**Reply by SilverStarCoder** [reply]
Thank you.
